src/refract/JSONSchemaVisitor.cc

```diff
@@ -278,45 +281,71 @@ namespace refract
         ObjectElement *o = new ObjectElement;
         ArrayElement::ValueType reqVals;
         std::vector<MemberElement*> varProps;
+        RefractElements oneOfMembers;
 
         o->renderType(IElement::rCompact);
 
         for (std::vector<refract::IElement*>::const_iterator it = val.begin();
              it != val.end();
              ++it) {
 
-            if (*it) {
-                MemberElement *mr = TypeQueryVisitor::as<MemberElement>(*it);
-                if (!mr) {
-                    continue;
-                }
+            if (!*it) {
+                continue;
+            }
 
-                if (IsTypeAttribute(*(*it), "required") ||
-                    IsTypeAttribute(*(*it), "fixed") ||
-                    (fixed && !IsTypeAttribute(*(*it), "optional"))) {
-                    StringElement *str = TypeQueryVisitor::as<StringElement>(mr->value.first);
-                    if (str) {
-                        reqVals.push_back(IElement::Create(str->value));
+            TypeQueryVisitor type;
+            type.visit(*(*it));
+
+            switch (type.get()) {
+                case TypeQueryVisitor::Member: {
+                    MemberElement *mr = static_cast<MemberElement*>(*it);
+                    if (IsTypeAttribute(*(*it), "required") ||
+                            IsTypeAttribute(*(*it), "fixed") ||
+                            (fixed && !IsTypeAttribute(*(*it), "optional"))) {
```

Comment:
It is not good idea use `drafter` namespace in `refract`.
Optionaly we can introduce "keyword library"  in refract (in future)